common/unsafe/src/main/java/org/apache/spark/unsafe/types/UTF8String.java

```diff
@@ -575,14 +575,14 @@ public UTF8String trim() {
   public UTF8String trimAll() {
     int s = 0;
     // skip all of the whitespaces (<=0x20) in the left side
-    while (s < this.numBytes && getByte(s) <= ' ') s++;
+    while (s < this.numBytes && canBeTrimmed(getByte(s))) s++;
```

Comment:
Does`java.lang.Character.isWhitespace(char)` work as well?

Reply:
yes, that's a good choice